### PR TITLE
DOC: add closed and dtype in TimedeltaIndex parameters

### LINKED ISSUE #49153: DOC: closed and dtype missing from TimedeltaIndex docs
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.TimedeltaIndex.html

### Documentation problem

Missing two parameters, closed and dtype

### Suggested fix for documentation

Add these two parameters with types ane explanation

xref pandas-dev/pandas-stubs#383

### PR DESCRIPTION
**Pull request by MarceloCMiguel**
- [x] closes #49153 
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/indexes/timedeltas.py [hunk 1]
```diff
@@ -60,6 +60,11 @@ class TimedeltaIndex(DatetimeTimedeltaMixin):
         One of pandas date offset strings or corresponding objects. The string
         'infer' can be passed in order to set the frequency of the index as the
         inferred frequency upon creation.
+    closed : str, default None
```
[the remark this answers is not in the document]
Comment (bashtage) [reply pandas/core/indexes/timedeltas.py:63]: Same was noticed in pandas-stubs by @Dr-Irv.  We thought deprecation or just removal made the most sense.

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/indexes/timedeltas.py:63]: agreed

[the remark this answers is not in the document]
Comment (MarceloCMiguel) [reply pandas/core/indexes/timedeltas.py:63]: Should I remove the closed parameter then?

[the remark this answers is not in the document]
Comment (bashtage) [reply pandas/core/indexes/timedeltas.py:63]: Ideally remove and deprecate closed using `@deprecate_nonkeyword_arguments`

[the remark this answers is not in the document]
Comment (MarceloCMiguel) [reply pandas/core/indexes/timedeltas.py:63]: I tried to find some other function that uses `@depreceate_kwarg` but couldn't find any, would it be something like this?
```
@deprecate_kwarg
closed : str, default None
    Make the interval closed with respect to the given frequency to
    the 'left', 'right', or both sides (None).
```

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/indexes/timedeltas.py:63]: You can reference the `@deprecate_kwarg` usage from this older PR: https://github.com/pandas-dev/pandas/pull/47912

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**